src/go/rpk/pkg/cli/cloud/mcp/mcp.go

```diff
@@ -209,3 +197,317 @@ func newStdioCommand(fs afero.Fs, p *config.Params) *cobra.Command {
 	cmd.Flags().BoolVarP(&allowDelete, "allow-delete", "", false, "Allow delete RPCs. Off by default")
 	return cmd
 }
+
+func newProxyCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var clusterID, mcpServerID string
+	var install bool
+	var mcpClient string
+	cmd := &cobra.Command{
+		Use:   "proxy",
+		Short: "Proxy requests to a remote Redpanda Cloud MCP server",
+		Long: `Proxy MCP requests to a remote Redpanda Cloud MCP server.
+
+This command connects to a specific cluster and MCP server running in that cluster,
+then proxies MCP requests from stdio to the remote MCP server over HTTP.
+
+Use --install to configure the MCP client instead of serving stdio.`,
+		Args: cobra.NoArgs,
+		PreRunE: func(_ *cobra.Command, _ []string) error {
+			if install && mcpClient == "" {
+				return fmt.Errorf("--client flag is required when using --install")
+			}
+			return nil
+		},
+		Run: func(_ *cobra.Command, _ []string) {
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "Failed to load config: %w", err)
+
+			// Start out with empty token, and use the maybeReloadToken function to update.
+			cl := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, "")
+			var m sync.RWMutex
+			var tokenOK bool
+
+			maybeReloadToken := func() error {
+				cfg, err := p.Load(fs)
+				if err != nil {
+					return err
+				}
+
```

Comment:
Why is this function needed? It is only used on L276 just after its declaration and it loads the profile again while it was loaded a few lines above.

Reply:
indeed, good catch. in stdio cmd, we use reload in a middleware to reload tokens, if needed, e.g. they ran rpk cloud login. here we don't do it, but we should. 

i updated it to have a middleware as well.